src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/index.tsx

```diff
@@ -305,39 +305,36 @@ class Filter extends Component<Props> {
                         )}
                       </Media>
                       <Media greaterThan="xs">
-                        <Sidebar width="25%" mr={2}>
-                          <Filters />
-                        </Sidebar>
+                        <Filters />
                       </Media>
+                    </Box>
+                    {/* Main Artwork Grid */}
+                    <Box width={["100%", "75%"]}>
+                      {!hideTopBorder && <Separator mb={2} mt={-1} />}
 
-                      {/* Main Artwork Grid */}
-                      <Box width={xs ? "100%" : "75%"}>
-                        {!hideTopBorder && <Separator mb={2} mt={-1} />}
+                      {this.renderSelect({
+                        filterState,
+                        mediator,
+                        xs,
+                      })}
 
-                        {this.renderSelect({
-                          filterState,
-                          mediator,
-                          xs,
-                        })}
-
-                        <Spacer mb={2} />
+                      <Spacer mb={2} />
 
-                        {this.showZeroState ? (
-                          this.renderZeroState({
-                            user,
-                            mediator,
-                          })
-                        ) : (
-                          <ArtworkFilter
-                            artist={this.props.artist}
-                            artistID={this.props.artist.id}
-                            columnCount={xs || sm || md ? 2 : 3}
-                            filters={filterState.state}
-                          />
-                        )}
-                      </Box>
-                    </Flex>
-                  </>
+                      {this.showZeroState ? (
+                        this.renderZeroState({
+                          user,
+                          mediator,
+                        })
+                      ) : (
+                        <ArtworkFilter
+                          artist={this.props.artist}
+                          artistID={this.props.artist.id}
+                          columnCount={xs || sm || md ? 2 : 3}
```

Comment:
I wonder if there's a responsive helper, like 
```
columnCount={responsive([2, 3, 4])}
```

Reply:
I’m actually wondering if we can’t just use `Media` for this too. Why did we think we couldn’t do that?

The only possible issue I see is having to on-the-fly calculate all the columns, but for that we could indeed make the count a responsive array, and then internally we just keep calculated layouts that we amend on each pagination. That way it’s probably even a bit faster than the current version (as currently it does the calculation on size change. The the problem is also abstracted away and we can work on it in the future in isolation.

What do you think?

Reply:
Well, we don’t even need to do the cache ahead optimization for the first step, just abstracting the issue by making the API responsive is enough so users can start doing the better thing.

Reply:
It seems like it would work, but I think that pretty soon we'll be using simple horizontal rows. At that point I'm not sure we'll be worrying about column counts and instead deferring to an artwork item width to define the layouts that we need -- 25% = 4, 33% = 3, 50% = 2, etc.

Reply:
Makes sense 👍